folium/folium.py

```diff
@@ -948,6 +950,40 @@ def json_style(style_cnt, line_color, line_weight, line_opacity,
         self.template_vars.setdefault('geo_styles', []).append(style)
         self.template_vars.setdefault('gjson_layers', []).append(layer)
 
+    @iter_obj('image_overlay')
+    def image_overlay(self, data, image_name=None, filename='data.png',
+                      lower_left=[-90,-180], upper_right=[90,180], opacity=0.25):
```

Comment:
I do not like that leaflet is alphabetical order `(lat, lon)` instead of geometrical order `(lon, lat)`.  Also, by aggregating into `lower_left` and `upper_right` tuples things might get even more confusing.

We have to write a good docstring describing the inputs to solve the confusion. But also consider using: `min_lat=-90, max_lat=90, min_lon=-180, max_lon=180`.  I know it is verbose, but it is unambiguous (and we already use it in the main class).